appengine/handleStatus.go

```diff
@@ -36,6 +39,13 @@ func handleStatus(c context.Context, w http.ResponseWriter, r *http.Request) err
 		return fmt.Errorf("No service account found for project %s: %v", project, err)
 	}
 
+	acmeTest := selfTest(c, r)
+	// The self test starts another request within the same go app. Because
+	// GOMAXPROCS is set to 1 on AppEngine, this causes a deadlock as the url
+	// fetch in this goroutine blocks and the other request never gets
+	// scheduled.
+	runtime.Gosched()
```

Comment:
Do the blocking urlfetch in a separate goroutine.  The `parallel.Parallel` call below already does a bunch of async operations in parallel on separate goroutines, just add this one in another func.

    var selfTest error
    if err := parallel.Parallel(nil, nil, func() error {
      ...
    }, func() error {
      acmeTest = selfTest(c, r)
      return nil
    }, ...